Compilation error

/home/wow/aznuevo/modules/mod-antifarming/src/cs_antifarming.cpp:58:25: error: no member named 'GetPlayerNameByGUID' in 'ObjectMgr'
            sObjectMgr->GetPlayerNameByGUID(itr->first, charName);
            ~~  ^
/home/wow/aznuevo/modules/mod-antifarming/src/cs_antifarming.cpp:59:45: error: no member named 'GetPlayerAccountIdByGUID' in 'ObjectMgr'
            AccountMgr::GetName(sObjectMgr->GetPlayerAccountIdByGUID(itr->first), accName);
                                ~~  ^

Compiled on ubuntu server 20.04, clang 10
AZ rev: b4a14214ca57611f023859c7d4a115b9815effb1

--- a/src/AntiFarming.cpp
+++ b/src/AntiFarming.cpp
@@ -37,7 +37,7 @@ public:
                     std::string str = "|cFFFFFC00[Anti-Farm System]|cFF00FFFF[|cFF60FF00" + killer->GetName() + "|cFF00FFFF] Possible Farmer!";
                     WorldPacket data(SMSG_NOTIFICATION, (str.size() + 1));
                     data << str;
-                    sWorld->SendGlobalGMMessage(&data);
+                    sWorldSessionMgr->SendGlobalGMMessage(&data);
 
                     // Kick player
                     if (sConfigMgr->GetOption<bool>("AntiFarming.KickPlayer", true))

--- a/src/AntiFarming.cpp
+++ b/src/AntiFarming.cpp
@@ -8,16 +8,20 @@
 
 class AntiFarmingPlayerScript : public PlayerScript
 {
-public: AntiFarmingPlayerScript() : PlayerScript("AntiFarmingPlayerScript") {}
+public:
+    AntiFarmingPlayerScript() : PlayerScript("AntiFarmingPlayerScript", {
+        PLAYERHOOK_ON_LOGIN,
+        PLAYERHOOK_ON_PVP_KILL
+    }) {}
 
-        void OnLogin(Player* player) override
+        void OnPlayerLogin(Player* player) override
         {
             if (sConfigMgr->GetOption<bool>("AntiFarming.Enable", true))
                 if (sConfigMgr->GetOption<bool>("AntiFarming.LoginMessage", true))
                     ChatHandler(player->GetSession()).PSendSysMessage("This server is running the |cff4CFF00Antifarming|r Module.");
         }
 
-        void OnPVPKill(Player* killer, Player* killed) override
+        void OnPlayerPVPKill(Player* killer, Player* killed) override
         {
             if (sConfigMgr->GetOption<bool>("AntiFarming.Enable", true))
             {

--- a/src/AntiFarming.cpp
+++ b/src/AntiFarming.cpp
@@ -6,25 +6,29 @@
 
 #include "AntiFarming.h"
 
-class AntiFarmingPlayerScript : public PlayerScript {
+class AntiFarmingPlayerScript : public PlayerScript
+{
 public: AntiFarmingPlayerScript() : PlayerScript("AntiFarmingPlayerScript") {}
 
-        void OnLogin(Player* player) override {
-            if (sConfigMgr->GetOption<bool>("AntiFarming.Enable", true)) {
+        void OnLogin(Player* player) override
+        {
+            if (sConfigMgr->GetOption<bool>("AntiFarming.Enable", true))
                 if (sConfigMgr->GetOption<bool>("AntiFarming.LoginMessage", true))
                     ChatHandler(player->GetSession()).PSendSysMessage("This server is running the |cff4CFF00Antifarming|r Module.");
-            }
         }
 
-        void OnPVPKill(Player* killer, Player* killed) override {
-            if (sConfigMgr->GetOption<bool>("AntiFarming.Enable", true)) {
+        void OnPVPKill(Player* killer, Player* killed) override
+        {
+            if (sConfigMgr->GetOption<bool>("AntiFarming.Enable", true))
+            {
                 uint32 KillerGUID = killer->GetGUID().GetCounter();
                 uint32 VictimGUID = killed->GetGUID().GetCounter();
 
                 if (KillerGUID == VictimGUID) // Suicide
                     return;
 
-                if (killer->GetSession()->GetRemoteAddress() == killed->GetSession()->GetRemoteAddress() || killed->GetMaxHealth() < (uint32)sConfigMgr->GetOption<uint32>("AntiFarming.MinHealth", 10000)) { // about 1/4 of the maximum hp a character can have on your server to make this fully functional.
+                if (killer->GetSession()->GetRemoteAddress() == killed->GetSession()->GetRemoteAddress() || killed->GetMaxHealth() < (uint32)sConfigMgr->GetOption<uint32>("AntiFarming.MinHealth", 10000)) // about 1/4 of the maximum hp a character can have on your server to make this fully functional.
+                {
                     sAntiFarming->dataMap[KillerGUID]++;
                     std::string str = "|cFFFFFC00[Anti-Farm System]|cFF00FFFF[|cFF60FF00" + killer->GetName() + "|cFF00FFFF] Possible Farmer!";
                     WorldPacket data(SMSG_NOTIFICATION, (str.size() + 1));
@@ -32,14 +36,16 @@ public: AntiFarmingPlayerScript() : PlayerScript("AntiFarmingPlayerScript") {}
                     sWorld->SendGlobalGMMessage(&data);
 
                     // Kick player
-                    if (sConfigMgr->GetOption<bool>("AntiFarming.KickPlayer", true)) {
+                    if (sConfigMgr->GetOption<bool>("AntiFarming.KickPlayer", true))
+                    {
                         uint32 kickWarnings = sConfigMgr->GetOption<uint32>("AntiFarming.KickPlayer.Warnings", 5);
                         if (sAntiFarming->dataMap[KillerGUID] >= kickWarnings)
                             killer->GetSession()->KickPlayer();
                     }
 
                     // Ban player
-                    if (sConfigMgr->GetOption<bool>("AntiFarming.BanPlayer", false)) {
+                    if (sConfigMgr->GetOption<bool>("AntiFarming.BanPlayer", false))
+                    {
                         std::string banTime = sConfigMgr->GetOption<std::string>("AntiFarming.BanTime", "1") + "d";
                         uint32 banWarnings = sConfigMgr->GetOption<uint32>("AntiFarming.BanPlayer.Warnings", 8);
                         if (sAntiFarming->dataMap[KillerGUID] >= banWarnings)
@@ -56,7 +62,8 @@ AntiFarming* AntiFarming::instance()
     return &instance;
 }
 
-void AddAntiFarmingScripts() {
+void AddAntiFarmingScripts()
+{
     new AntiFarmingPlayerScript();
     AddSC_antifarming_commandscript();
 }

--- a/src/AntiFarming.h
+++ b/src/AntiFarming.h
@@ -1,17 +1,18 @@
 ﻿#ifndef ANTIFARMING_H
 #define ANTIFARMING_H
 
-#include "ScriptMgr.h"
-#include "Player.h"
+#include "AccountMgr.h"
 #include "BanMgr.h"
+#include "Chat.h"
+#include "Configuration/Config.h"
+#include "cs_antifarming.h"
 #include "Language.h"
 #include "ObjectMgr.h"
-#include "Chat.h"
-#include "AccountMgr.h"
-#include "cs_antifarming.h"
-#include "Configuration/Config.h"
+#include "Player.h"
+#include "ScriptMgr.h"
 
-class AntiFarming {
+class AntiFarming
+{
 public:
     static AntiFarming* instance();
     typedef std::map<uint64, uint8> antiFarmingData;

--- a/src/AntiFarming.h
+++ b/src/AntiFarming.h
@@ -10,6 +10,7 @@
 #include "ObjectMgr.h"
 #include "Player.h"
 #include "ScriptMgr.h"
+#include "WorldSessionMgr.h"
 
 class AntiFarming
 {

--- a/src/cs_antifarming.cpp
+++ b/src/cs_antifarming.cpp
@@ -1,10 +1,12 @@
 ﻿#include "AntiFarming.h"
 
-class antifarming_commandscript : public CommandScript {
+class antifarming_commandscript : public CommandScript
+{
 public:
     antifarming_commandscript() : CommandScript("antifarming_commandscript") {}
 
-    std::vector<ChatCommand> GetCommands() const override {
+    std::vector<ChatCommand> GetCommands() const override
+    {
 
         static std::vector<ChatCommand> HelpDeleteSubCommandTable = {
             { "all",		SEC_GAMEMASTER,		false,	&HandleHelpDeleteAllCommand,	"" },
@@ -35,7 +37,8 @@ public:
         return commandTable;
     }
 
-    static bool HandleLogCommand(ChatHandler* handler, const char* args) {
+    static bool HandleLogCommand(ChatHandler* handler, const char* args)
+    {
         WorldSession *Session = handler->GetSession();
         uint32 RLimit = 10;
         uint32 i = 0;
@@ -45,7 +48,8 @@ public:
         if (RLimit == 0 || RLimit > 100)
             RLimit = 10;
 
-        if (sAntiFarming->dataMap.empty()) {
+        if (sAntiFarming->dataMap.empty())
+        {
             Session->SendNotification("There are no records");
             return false;
         }
@@ -54,7 +58,8 @@ public:
         std::string charName;
         std::string accName;
 
-        for (AntiFarming::antiFarmingData::iterator itr = sAntiFarming->dataMap.begin(); itr != sAntiFarming->dataMap.end() && i < RLimit; ++itr, i++) {
+        for (AntiFarming::antiFarmingData::iterator itr = sAntiFarming->dataMap.begin(); itr != sAntiFarming->dataMap.end() && i < RLimit; ++itr, i++)
+        {
             sCharacterCache->GetCharacterNameByGuid(ObjectGuid(itr->first), charName);
             AccountMgr::GetName(sCharacterCache->GetCharacterAccountIdByGuid(ObjectGuid(itr->first)), accName);
             snprintf(msg, 250, "ID: |cFFFFFFFF%lu|r | Character: |cFFFFFFFF%s|r | Account: |cFFFFFFFF%s|r | Warning Level: |cFFFF0000%u|r\n", (long)itr->first, charName.c_str(), accName.c_str(), itr->second);
@@ -64,14 +69,17 @@ public:
         return true;
     }
 
-    static bool HandleDeleteAllCommand(ChatHandler* handler, const char* /*args*/) {
+    static bool HandleDeleteAllCommand(ChatHandler* handler, const char* /*args*/)
+    {
         WorldSession *Session = handler->GetSession();
 
-        if (sAntiFarming->dataMap.empty()) {
+        if (sAntiFarming->dataMap.empty())
+        {
             Session->SendNotification("There are no records!");
             return false;
         }
-        else {
+        else
+        {
             sAntiFarming->dataMap.clear();
             Session->SendAreaTriggerMessage("All records were successfully deleted!");
             return true;
@@ -79,13 +87,15 @@ public:
         return true;
     }
 
-    static bool HandleDeleteIDCommand(ChatHandler* handler, const char* args) {
+    static bool HandleDeleteIDCommand(ChatHandler* handler, const char* args)
+    {
         if (!*args)
             return false;
         WorldSession *Session = handler->GetSession();
         uint32 id = atoi((char*)args);
         AntiFarming::antiFarmingData::iterator it = sAntiFarming->dataMap.find(id);
-        if (it == sAntiFarming->dataMap.end()) {
+        if (it == sAntiFarming->dataMap.end())
+        {
             char msg[250];
             snprintf(msg, 250, "Log with ID \"%u\" doesn't exist!", id);
             Session->SendNotification("%s", msg);
@@ -96,7 +106,8 @@ public:
         return true;
     }
 
-    static bool HandleHelpLogCommand(ChatHandler* handler, const char* /*args*/) {
+    static bool HandleHelpLogCommand(ChatHandler* handler, const char* /*args*/)
+    {
         handler->PSendSysMessage("This command allows you to check the Database Log for abusers of the Anti-Farm System \n");
         handler->PSendSysMessage("Entering a number after this command will be used to check the maximum lines shown \n");
         handler->PSendSysMessage("Example : \".afs log 15\" will show the first 15 records, even if there are more.");
@@ -104,14 +115,16 @@ public:
         return true;
     }
 
-    static bool HandleHelpDeleteAllCommand(ChatHandler* handler, const char* /*args*/) {
+    static bool HandleHelpDeleteAllCommand(ChatHandler* handler, const char* /*args*/)
+    {
         handler->PSendSysMessage("This command will delete all existing records of abusers \n");
         handler->PSendSysMessage("This command does not have any arguments.");
         handler->SetSentErrorMessage(true);
         return true;
     }
 
-    static bool HandleHelpDeleteIDCommand(ChatHandler* handler, const char* /*args*/) {
+    static bool HandleHelpDeleteIDCommand(ChatHandler* handler, const char* /*args*/)
+    {
         handler->PSendSysMessage("This command will let you delete a specific record from the Database \n");
         handler->PSendSysMessage("Entering a number after this command will act like the ID of the record \n");
         handler->PSendSysMessage("Example : \".afs delete ID 50 \" will delete the record with ID 50.");
@@ -120,6 +133,7 @@ public:
     }
 };
 
-void AddSC_antifarming_commandscript() {
+void AddSC_antifarming_commandscript()
+{
     new antifarming_commandscript();
 }

--- a/src/cs_antifarming.cpp
+++ b/src/cs_antifarming.cpp
@@ -1,5 +1,7 @@
 ﻿#include "AntiFarming.h"
 
+using namespace Acore::ChatCommands;
+
 class antifarming_commandscript : public CommandScript
 {
 public:

--- a/src/cs_antifarming.cpp
+++ b/src/cs_antifarming.cpp
@@ -5,47 +5,47 @@ class antifarming_commandscript : public CommandScript
 public:
     antifarming_commandscript() : CommandScript("antifarming_commandscript") {}
 
-    std::vector<ChatCommand> GetCommands() const override
+    ChatCommandTable GetCommands() const override
     {
 
-        static std::vector<ChatCommand> HelpDeleteSubCommandTable = {
-            { "all",		SEC_GAMEMASTER,		false,	&HandleHelpDeleteAllCommand,	"" },
-            { "ID",		    SEC_GAMEMASTER,		false,	&HandleHelpDeleteIDCommand,		"" }
-        };
-
-        static std::vector<ChatCommand> HelpCommandSubTable = {
-            { "log",		SEC_GAMEMASTER,		false,	&HandleHelpLogCommand,		"" },
-            { "delete",		SEC_GAMEMASTER,		false,	NULL,		            	"",	HelpDeleteSubCommandTable }
-        };
-
-        static std::vector<ChatCommand> DeleteCommandSubTable = {
-            { "all",		SEC_GAMEMASTER,		true,	&HandleDeleteAllCommand,	"" },
-            { "ID",		    SEC_GAMEMASTER,		true,	&HandleDeleteIDCommand,		"" }
-        };
-
-        static std::vector<ChatCommand> AFSSubCommandTable = {
-            { "log",		SEC_GAMEMASTER,		true,	&HandleLogCommand,		    "" },
-            { "delete",		SEC_ADMINISTRATOR,	true,	NULL,		            	"", DeleteCommandSubTable },
-            { "help",		SEC_GAMEMASTER,		false,	NULL,		            	"", HelpCommandSubTable }
-        };
-
-
-        static std::vector<ChatCommand> commandTable =
+        static ChatCommandTable HelpDeleteSubCommandTable =
         {
-            { "afs",	    SEC_GAMEMASTER,		true,	NULL,		            	"",	AFSSubCommandTable }
+            { "all", HandleHelpDeleteAllCommand, SEC_GAMEMASTER, Console::No },
+            { "ID",  HandleHelpDeleteIDCommand,  SEC_GAMEMASTER, Console::No }
+        };
+
+        static ChatCommandTable HelpCommandSubTable =
+        {
+            { "log",    HandleHelpLogCommand, SEC_GAMEMASTER, Console::No },
+            { "delete", HelpDeleteSubCommandTable }
+        };
+
+        static ChatCommandTable DeleteCommandSubTable =
+        {
+            { "all", HandleDeleteAllCommand, SEC_ADMINISTRATOR, Console::Yes },
+            { "ID",  HandleDeleteIDCommand,  SEC_ADMINISTRATOR, Console::Yes }
+        };
+
+        static ChatCommandTable AFSSubCommandTable =
+        {
+            { "log",    HandleLogCommand, SEC_GAMEMASTER, Console::Yes },
+            { "delete", DeleteCommandSubTable },
+            { "help",   HelpCommandSubTable }
+        };
+
+        static ChatCommandTable commandTable =
+        {
+            { "afs", AFSSubCommandTable }
         };
         return commandTable;
     }
 
-    static bool HandleLogCommand(ChatHandler* handler, const char* args)
+    static bool HandleLogCommand(ChatHandler* handler, std::optional<uint32> RLimit)
     {
         WorldSession *Session = handler->GetSession();
-        uint32 RLimit = 10;
         uint32 i = 0;
-        if ((char*)args)
-            RLimit = atoi((char*)args);
 
-        if (RLimit == 0 || RLimit > 100)
+        if (!RLimit.has_value() || RLimit == 0 || RLimit > 100)
             RLimit = 10;
 
         if (sAntiFarming->dataMap.empty())
@@ -69,7 +69,7 @@ public:
         return true;
     }
 
-    static bool HandleDeleteAllCommand(ChatHandler* handler, const char* /*args*/)
+    static bool HandleDeleteAllCommand(ChatHandler* handler)
     {
         WorldSession *Session = handler->GetSession();
 
@@ -87,12 +87,9 @@ public:
         return true;
     }
 
-    static bool HandleDeleteIDCommand(ChatHandler* handler, const char* args)
+    static bool HandleDeleteIDCommand(ChatHandler* handler, uint32 id)
     {
-        if (!*args)
-            return false;
         WorldSession *Session = handler->GetSession();
-        uint32 id = atoi((char*)args);
         AntiFarming::antiFarmingData::iterator it = sAntiFarming->dataMap.find(id);
         if (it == sAntiFarming->dataMap.end())
         {
@@ -106,7 +103,7 @@ public:
         return true;
     }
 
-    static bool HandleHelpLogCommand(ChatHandler* handler, const char* /*args*/)
+    static bool HandleHelpLogCommand(ChatHandler* handler)
     {
         handler->PSendSysMessage("This command allows you to check the Database Log for abusers of the Anti-Farm System \n");
         handler->PSendSysMessage("Entering a number after this command will be used to check the maximum lines shown \n");
@@ -115,7 +112,7 @@ public:
         return true;
     }
 
-    static bool HandleHelpDeleteAllCommand(ChatHandler* handler, const char* /*args*/)
+    static bool HandleHelpDeleteAllCommand(ChatHandler* handler)
     {
         handler->PSendSysMessage("This command will delete all existing records of abusers \n");
         handler->PSendSysMessage("This command does not have any arguments.");
@@ -123,7 +120,7 @@ public:
         return true;
     }
 
-    static bool HandleHelpDeleteIDCommand(ChatHandler* handler, const char* /*args*/)
+    static bool HandleHelpDeleteIDCommand(ChatHandler* handler)
     {
         handler->PSendSysMessage("This command will let you delete a specific record from the Database \n");
         handler->PSendSysMessage("Entering a number after this command will act like the ID of the record \n");

--- a/src/cs_antifarming.cpp
+++ b/src/cs_antifarming.cpp
@@ -44,7 +44,6 @@ public:
 
     static bool HandleLogCommand(ChatHandler* handler, std::optional<uint32> RLimit)
     {
-        WorldSession *Session = handler->GetSession();
         uint32 i = 0;
 
         if (!RLimit.has_value() || RLimit == 0 || RLimit > 100)
@@ -52,7 +51,7 @@ public:
 
         if (sAntiFarming->dataMap.empty())
         {
-            Session->SendNotification("There are no records");
+            handler->SendNotification("There are no records");
             return false;
         }
 
@@ -65,7 +64,7 @@ public:
             sCharacterCache->GetCharacterNameByGuid(ObjectGuid(itr->first), charName);
             AccountMgr::GetName(sCharacterCache->GetCharacterAccountIdByGuid(ObjectGuid(itr->first)), accName);
             snprintf(msg, 250, "ID: |cFFFFFFFF%lu|r | Character: |cFFFFFFFF%s|r | Account: |cFFFFFFFF%s|r | Warning Level: |cFFFF0000%u|r\n", (long)itr->first, charName.c_str(), accName.c_str(), itr->second);
-            handler->PSendSysMessage("%s", msg);
+            handler->PSendSysMessage("{}", msg);
             handler->SetSentErrorMessage(true);
         }
         return true;
@@ -77,7 +76,7 @@ public:
 
         if (sAntiFarming->dataMap.empty())
         {
-            Session->SendNotification("There are no records!");
+            handler->SendNotification("There are no records!");
             return false;
         }
         else
@@ -97,7 +96,7 @@ public:
         {
             char msg[250];
             snprintf(msg, 250, "Log with ID \"%u\" doesn't exist!", id);
-            Session->SendNotification("%s", msg);
+            handler->SendNotification("{}", msg);
             return false;
         }
         sAntiFarming->dataMap.erase(it);

